src/FileIndexer.ts

```diff
@@ -397,6 +443,81 @@ export class FileIndexer {
     }
     return node.getText() + ': ' + type()
   }
+
+  // Invokes the `onParent` callback for all "parents" of the provided node,
+  // where "parent" is loosely defined as the superclass or superinterface of
+  // that node. The callback is invoked on the `node` parameter itself if it's
+  // class-like or an interface.
+  private forEachParent(
+    node: ts.Node,
+    onParent: (
+      parent: ts.ClassLikeDeclaration | ts.InterfaceDeclaration
+    ) => void
+  ): void {
+    const isVisited = new Set<ts.Node>()
+    const loop = (declaration: ts.Node): void => {
+      if (isVisited.has(declaration)) {
+        return
+      }
+      isVisited.add(declaration)
+      if (
+        ts.isClassLike(declaration) ||
+        ts.isInterfaceDeclaration(declaration)
+      ) {
+        onParent(declaration)
+      }
+      if (ts.isObjectLiteralExpression(declaration)) {
+        const tpe = this.inferredTypeOfObjectLiteral(declaration)
+        for (const symbolDeclaration of tpe.symbol?.declarations || []) {
+          loop(symbolDeclaration)
+        }
+      } else if (ts.isClassLike(declaration)) {
+        for (const heritageClause of declaration?.heritageClauses || []) {
+          for (const tpe of heritageClause.types) {
+            const parentSymbol = this.getTSSymbolAtLocation(tpe.expression)
+            if (parentSymbol) {
+              for (const parentDecl of parentSymbol.declarations || []) {
+                loop(parentDecl)
+              }
+            }
+          }
+        }
```

Comment:
Could you add a TODO here that we should handle the parents of interfaces too? (And briefly mention which other other cases are missing.)

I'm guessing this is what you are referring to in the PR description

> This PR does not implement all locations where inheritance is used, but we can add that functionality later.

Reply:
I added an `|| ts.isInterfaceDeclaration(declaration)` so that we recursively traverse both class and interface parents. I don't have any examples at the top of my mind that are not handled besides object literals that implement interfaces need an occurrence (local symbol) so that "find implementation" on the interface symbol itself goes to the object literal. I opened an issue to track this case https://github.com/sourcegraph/scip-typescript/issues/138